src/assembler/mod.rs

```diff
@@ -49,8 +46,10 @@ pub fn assemble<'a, W: Write>(input: &'a str, output: &mut W) -> AssembleResult<
     }
 }
 
+type ParsingResult = Result<(), String>;
+
 fn process_tokens<'a, W: Write>(
-    input: &'a str,
+    _input: &'a str,
```

Comment:
Any reason for keeping the `_input` parameter around if it's not used?

Reply:
We need lifetime of this parameter for `AssembleResult<'a>`, but now I realized that we might just return new `ParsingResult` here as well. Made a new commit, now `input` used only in `assemble`